oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -123,10 +118,13 @@ public String getIndexPrefix() {
     /**
      * Checks if elastic server is available for connection.
      * @return true if available, false otherwise.
-     * @throws IOException if some connection exception occurs.
      */
-    public boolean isAvailable() throws IOException {
-        return this.getClient().ping(RequestOptions.DEFAULT);
+    public boolean isAvailable() {
+        try {
+            return this.getClient().ping(RequestOptions.DEFAULT);
```

Comment:
If the client is closed here already(not sure if that can be the case), then this method would throw IllegalStateException and not return false. Maybe we can handle that exception as well. Also if we would be handling different exceptions - might make sense to log a warn message in the catch for both of them here itself.

Reply:
I have added some log calls in the catch clause. I would not catch the `IllegalStateException` because in this case (the client is closed before the check) it would be not correct to return `false` since we were not able to test the connection. In general, I would tend to not catch `RuntimeException`.

Reply:
makes sense.